### PR TITLE
Implement  Script to Seed Firestore with Monument Data for New Contributors✅

### PR DESCRIPTION
**Pull request by kareemabdeen**
## Description

This PR introduces a standalone JavaScript script to populate the Firestore database with monument data, eliminating the need to run or modify the Flutter app. Additionally, it updates the README to guide new contributors through the process of seeding Firestore, ensuring they can easily get started without altering the core app code.

## Approach Steps

- Created a dedicated JavaScript script (`populate_monuments.js`) that:
  - Uses the Firebase Admin SDK for authentication and Firestore access.
  - Populates the Firestore `monuments` collection with predefined data.
- Updated the README with step-by-step instructions on:
  - Installing Node.js and the Firebase Admin SDK.
  - Obtaining a service account key.
  - Running the script to seed data into Firestore.

These changes simplify the onboarding experience for new contributors and provide a reliable, one-time data seeding process independent of the main app’s codebase.

  - Fixed `RangeError` in the `ImageTile` widget:
  - Updated rendering logic to dynamically generate tiles using `List.generate` based on the length of the `images` list.
  - Prevents out-of-bounds errors and ensures robustness.

This improves the app's ability to handle monument data efficiently and enhances the stability of the image display components.

**Fixes**: #181

---

## Type of change

Please select the relevant options:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (does not change functionality, e.g., code style improvements, linting)
- [ ] Documentation update

---

## How Has This Been Tested?

1. **Local Testing of the Script**:  
   - Ran `node populate_monuments.js` in a local environment.
   - Verified successful console output indicating that the `monuments` collection was populated.

2. **Firestore Verification**:  
   - Checked the Firebase console to confirm that the `monuments` collection appeared and that the document fields matched the predefined data.

3. **No App Code Required**:  
   - Ensured that the script functioned without running or modifying the Flutter app.
   - Confirmed that future runs of the script (with updated data) also correctly reflected changes in the Firestore collection.

- **ImageTile Widget**:
  - Tested rendering with various `images` list lengths to confirm no `RangeError` occurs.
  - Verified that the UI dynamically adapts to the available number of images.

### Screenshots

photo_2024-12-12_23-27-35

photo_2024-12-12_23-27-31

https://github.com/user-attachments/assets/533c29ae-f7b1-4824-80f4-64883863edde

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I have checked my code and corrected any misspellings.

---

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Tag the PR with the appropriate labels

### CODE REVIEW

**Comment by kareemabdeen** [comment]
Can you please check this PR? @mdmohsin7

**Comment by mdmohsin7** [comment]
Can we have something else instead? Having stuff like this in the app is a bit of an issue as it isn't really meant for the app. It has to be something a user can run once to save the details in firestore without even needing to setup the app, that's all. A simple python or any script would do

**Comment by mdmohsin7** [comment]
Also pls open separate PRs for issues/fixes that are different from the core issue addressed by the PR

**Comment by kareemabdeen** [comment]
Ok, Consider it dn insha'Allah

**Comment by kareemabdeen** [comment]
Hey @mdmohsin7 ,

Could you please review the current changes? I’ve also included the fix for the image tile exception in the same PR. This allows me to capture a screenshot and record a video for the PR details, making it easier for both of us to visualize and evaluate the results. Thanks for your time!

**Comment by mdmohsin7** [comment]
Nice! @kareemabdeen can you please create a new folder named `scripts` in this repository at the root level and add the js file there? Once that is done, I'll merge the PR

**Comment by kareemabdeen** [comment]
All is dn, can you please review new changes? @mdmohsin7

**Comment by mdmohsin7** [comment]
@kareemabdeen can you please also update the readme file steps regarding the script

**Comment by kareemabdeen** [comment]
I am already added the relevant steps after making current changes , mentioned in last commit in this branch 
@mdmohsin7 

image

File: README.md [hunk 1]
```diff
@@ -63,7 +63,57 @@ Follow the official [Firebase guide](https://firebase.google.com/docs/flutter/se
...
+- Save `serviceAccountKey.json` in a folder dedicated to running the script (not necessarily inside the Monumento project folder).
+
+### Steps to Populate Data
+
+1. **Set Up a Separate Folder**:
+- Create a new folder (for example `firestore_scripts`) outside of the app directory, such as on your desktop, to organize and manage Firestore-related   scripts independently.
+
+2. **Add the Script and Key**:
+- Place your `serviceAccountKey.json` here.
+- First, create a file named populate_monuments.js in the firestore_scripts folder you created outside of the app directory (e.g., on your desktop).
```
Comment (mdmohsin7) [inline README.md]: I think these steps are not needed anymore, instead we should tell the contributors to simply place the json file in the scripts folder and then run the command from that directory

**Comment by mdmohsin7** [comment]
Also pls add an entry to gitignore for `scripts/serviceAccountKey.json` so that no one commits it by mistake

**Comment by kareemabdeen** [comment]
All is dn
branch ready to be merged @mdmohsin7

**Review by mdmohsin7 — APPROVED** [review]
lgtm

**Pull request merged**